src/main/java/com/example/demo/products/AutoFire.java

```diff
@@ -0,0 +1,5 @@
+package com.example.demo.products;
+
+public interface AutoFire {
+    Integer rateOfFire = null;
```

Comment:
it is already null, without explicitly stating that

why would you use fields in interfaces? a lil cringy, isn't it?